### PR TITLE
checks whether correct params for mobile have been passed during OTP request

### LINKED ISSUE #300: Under OTP request, add check for the case when mobile has not been given

### PR DESCRIPTION
**Pull request by gurleen3**
Fixes #300

## Summary
Checks whether correct params have been passed during OTP request 

Tests are written for testing whether the correct params(mobile) have been passed while requesting otp. 

## Test Plan

- [x] Wrote tests
- [x] Tested locally
- [ ] Tested on staging
- [ ] Tested on production
- [ ] ~If adding any new environment variable:~
    - [ ] update `docs/ENV.md`
    - [ ] update the environment variables for staging and production
- [ ] ~If changes in DB, update [DB schema](https://docs.google.com/presentation/d/15c45HbHphh4w3guaFV3BgEA5nMKd-4J35JdUKdrpPzo/edit#slide=id.gc8c9ec0b5e_0_0) and BigQuery (staging and prod)~

### CODE REVIEW

File: users/tests.py [hunk 1]
```diff
@@ -11,13 +11,18 @@
...
 
-class OtpAuthTestCase(BaseTestCase):
+class OtpAuthTestCase(BaseTestCase):  
     """Tests the OTP functionality."""
 
     def setUp(self):
         super().setUp()
         # unset client credentials token so that the subsequent API calls goes as guest
         self.client.credentials()
+    def test_request_otp_mobile_not_passed(self):
```
Comment (dalmia) [inline users/tests.py:21]: all functions within classes should be separate by a space. Currently, there is no space before this function.

Comment (dalmia) [inline users/tests.py:21]: leaving a few similar feedback points as I mentioned on Manya's PR: the name of the test should specify what should happen given a particular set of inputs. So, in this case, a better name could be: `test_requesting_otp_fails_when_mobile_not_passed`

File: users/tests.py [hunk 2]
```diff
@@ -11,13 +11,18 @@
...
+        response = self.client.post(
+            reverse("request-otp")
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```
Comment (dalmia) [inline users/tests.py:25]: do check for the error message as well - the way Manya has done in her PR: https://github.com/avantifellows/plio-backend/pull/304

**Review by dalmia — CHANGES_REQUESTED** [review]

File: users/tests.py [hunk 3]
```diff
@@ -11,13 +11,18 @@
 from plio.cache import get_cache_key
 
 
-class OtpAuthTestCase(BaseTestCase):
+class OtpAuthTestCase(BaseTestCase):
```
Comment (deepansh96) [inline users/tests.py:14]: There's an extra space character added here
Usually it's best to see the "Files Changed" tab in a pull request yourself so you can see what all got changed from your commits

Comment (deepansh96) [reply users/tests.py:14]: I guess pre commit is also failing. All the spacing issues should be fixed by pre commit

Comment (gurleen3) [reply users/tests.py]: done

File: users/tests.py (hunk 2 above)
Comment (gurleen3) [reply users/tests.py:24]: corrected

File: users/tests.py (hunk 1 above)
Comment (gurleen3) [reply users/tests.py]: corrected it

**Pull request merged**